src/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/DisplayFormatAttribute.cs

```diff
@@ -45,5 +77,53 @@ public DisplayFormatAttribute()
         ///     Gets or sets a value indicating whether the field should be html encoded
         /// </summary>
         public bool HtmlEncode { get; set; }
+
+        /// <summary>
+        ///     Gets or sets the <see cref="System.Type" /> that contains the resources for <see cref="NullDisplayText" />.
+        ///     Using <see cref="NullDisplayTextResourceType" /> along with this property, allows the <see cref="GetNullDisplayText" />
+        ///     method to return localized values.
+        /// </summary>
+        public Type NullDisplayTextResourceType
+        {
+            get { return resourceType; }
+            set
+            {
+                if (resourceType != value)
+                {
+                    resourceType = value;
+                    nullDisplayText.ResourceType = value;
+                }
+            }
+        }
+
+        /// <summary>
+        ///     Gets the UI display string for NullDisplayText.
+        ///     <para>
+        ///         This can be either a literal, non-localized string provided to <see cref="NullDisplayText" /> or the
+        ///         localized string found when <see cref="NullDisplayTextResourceType" /> has been specified and <see cref="NullDisplayText" />
+        ///         represents a resource key within that resource type.
+        ///     </para>
+        /// </summary>
+        /// <returns>
+        ///     When <see cref="NullDisplayTextResourceType" /> has not been specified, the value of
+        ///     <see cref="NullDisplayText" /> will be returned.
+        ///     <para>
+        ///         When <see cref="NullDisplayTextResourceType" /> has been specified and <see cref="NullDisplayText" />
+        ///         represents a resource key within that resource type, then the localized value will be returned.
+        ///     </para>
+        ///     <para>
+        ///         Can return <c>null</c> and will not fall back onto other values, as it's more likely for the
+        ///         consumer to want to fall back onto the property name.
```

Comment:
Consumers are highly unlikely to fall back to the property name. Just say `When <see cref="NullDisplayText" /> and <see cref="NullDisplayTextResourceType" /> have not been set, returns <c>null</c>.`